accounts/abi/abigen/bindv2.go

```diff
@@ -0,0 +1,376 @@
+}
+
+func newContractBinder(binder *binder) *contractBinder {
+	return &contractBinder{
+		binder,
+		make(map[string]*tmplMethod),
+		make(map[string]*tmplEvent),
+		make(map[string]*tmplError),
+		make(map[string]bool),
+		make(map[string]bool),
+		make(map[string]bool),
+	}
+}
+
+// registerIdentifier applies alias renaming, name normalization (conversion
+// from snake to camel-case), and registers the normalized name in the specified identifier map.
+// It returns an error if the normalized name already exists in the map.
+func (cb *contractBinder) registerIdentifier(identifiers map[string]bool, original string) (normalized string, err error) {
+	normalized = abi.ToCamelCase(alias(cb.binder.aliases, original))
+
+	// Name shouldn't start with a digit. It will make the generated code invalid.
+	if len(normalized) > 0 && unicode.IsDigit(rune(normalized[0])) {
+		normalized = fmt.Sprintf("E%s", normalized)
+		normalized = abi.ResolveNameConflict(normalized, func(name string) bool {
+			_, ok := identifiers[name]
+			return ok
+		})
+	}
+	if _, ok := identifiers[normalized]; ok {
+		return "", fmt.Errorf("duplicate symbol '%s'", normalized)
+	}
+	identifiers[normalized] = true
+	return normalized, nil
+}
+
+// bindMethod registers a method to be emitted in the bindings. The name, inputs
+// and outputs are normalized. If any inputs are struct-type their structs are
+// registered to be emitted in the bindings. Any methods that return more than
+// one output have their results gathered into a struct.
+func (cb *contractBinder) bindMethod(original abi.Method) error {
+	normalized := original
+	normalizedName, err := cb.registerIdentifier(cb.callIdentifiers, original.Name)
+	if err != nil {
+		return err
+	}
+	normalized.Name = normalizedName
+
+	normalized.Inputs = normalizeArgs(original.Inputs)
+	for _, input := range normalized.Inputs {
+		if hasStruct(input.Type) {
+			cb.binder.BindStructType(input.Type)
+		}
+	}
+	normalized.Outputs = normalizeArgs(original.Outputs)
+	for j, output := range normalized.Outputs {
+		if output.Name != "" {
+			normalized.Outputs[j].Name = abi.ToCamelCase(output.Name)
+		}
+		if hasStruct(output.Type) {
+			cb.binder.BindStructType(output.Type)
+		}
+	}
+	isStructured := structured(original.Outputs)
+
+	// If the call returns multiple values, gather them into a struct
+	if len(normalized.Outputs) > 1 {
+		isStructured = true
+	}
+	cb.calls[original.Name] = &tmplMethod{
+		Original:   original,
+		Normalized: normalized,
+		Structured: isStructured,
+	}
+	return nil
+}
+
+// normalize a set of arguments by stripping underscores, giving a generic name
+// in the case where the arg name collides with a reserved Go keyword, and finally
+// converting to camel-case.
+func normalizeArgs(args abi.Arguments) abi.Arguments {
+	args = slices.Clone(args)
+	used := make(map[string]bool)
+
+	for i, input := range args {
+		if isKeyWord(input.Name) {
+			args[i].Name = fmt.Sprintf("Arg%d", i)
```

Comment:
Any particular reason to use the capitalized argument name?

e.g.

```go
func (erc20 *Erc20) PackAllowance(Arg0 common.Address, Arg1 common.Address) []byte {
	enc, err := erc20.abi.Pack("allowance", Arg0, Arg1)
	if err != nil {
		panic(err)
	}
	return enc
}
```

Reply:
The arg names are used to assign field names in the structs generated from function outputs.  But I've changed it so that the arg names are lowercase-prefixed, and are capitalized in the template where they are used as public fields of generated structs.